### PR TITLE
feat: expose core db connection settings

### PR DESCRIPTION
**Pull request by LiuShuaiyi**
Allow to change core.spec.database.maxIdleConnections and core.spec.database.maxOpenConnections in harborcluster.

### CODE REVIEW

File: apis/goharbor.io/v1beta1/harbor_types.go
Comment (MarcelMue) [inline apis/goharbor.io/v1beta1/harbor_types.go:381]: Why is this set here and not in `HarborDatabaseSpec`?

Comment (LiuShuaiyi) [reply apis/goharbor.io/v1beta1/harbor_types.go:381]: This is passed to `Core` and eventually Harbor core deployment to configure the max connections between `core` pods and the db, ultimately goes to https://sourcegraph.com/github.com/goharbor/harbor@cbb211e6702517fddcc8b9a7341e8a3479211ca7/-/blob/src/common/dao/pgsql.go?L99.

This is different from the configuring DB's server-side max connections.

Therefore I think this should be in harbor_types instead of in `HarborDatabaseSpec`. Thoughts?

**Review by thcdrt — APPROVED** [review]
LGTM

**Review by MarcelMue — APPROVED** [review]

**Comment by LiuShuaiyi** [comment]
Please don't merge yet (hopefully some of the checks will break). It's missing at least `charts/harbor-operator/templates/crds.yaml`

**Comment by thcdrt** [comment]
@MarcelMue @wy65701436 @chlins should we merge it now ? Or do we wait to recreate 1.4 release branch first ?

**Comment by MarcelMue** [comment]
> @MarcelMue @wy65701436 @chlins should we merge it now ? Or do we wait to recreate 1.4 release branch first ?

I think it's fine to merge now. Also why is CI stuck so weirdly? I am uncertain why github actions are not running.

**Comment by thcdrt** [comment]
> > @MarcelMue @wy65701436 @chlins should we merge it now ? Or do we wait to recreate 1.4 release branch first ?
> 
> I think it's fine to merge now. Also why is CI stuck so weirdly? I am uncertain why github actions are not running.

I just asked me the same question in another PR, I don't understand neither... @wy65701436 @chlins do you have an idea ?

**Comment by LiuShuaiyi** [comment]
Gentle ping @chlins @holyhope. Looks like I need an approval from a maintainer to trigger the workflows.

**Comment by LiuShuaiyi** [comment]
The PR failed at test the following test:

```
Test Case - Tag Immutability                                          Filesystem      Size  Used Avail Use% Mounted on
overlay          84G   76G  7.4G  92% /
tmpfs            64M     0   64M   0% /dev
shm              64M     0   64M   0% /dev/shm
/dev/root        84G   76G  7.4G  92% /drone
none            3.4G     0  3.4G   0% /tmp
| FAIL |
1 != 0
```

I don't think this failure is related to my change, right? Is it possible to re-run the test or merge it?

**Comment by OrlinVasilev** [comment]
@LiuShuaiyi congrats on merging your first PR here!

**Pull request merged**
